### PR TITLE
Shaded Hills features and fixes

### PR DESCRIPTION
**Pull request by noelle-lavenza**
## Description of changes
- Fix reagent overlay colours
- Fix object edibility checks (they were only checking if you could drink even if you were eating)
- Adds a filling colour for chopped meat
- Gives butchery items flavours other than 'umami'
- Fixes stack and prayer beads gender (stack gender is NEUTER if 1 piece and PLURAL if 2 or more, prayer beads is always PLURAL)
- Fixes loot piles being infinitely lootable even if re-looting is disabled
- Renames the champagne bottle in the fantasy modpack.
- Renames the white wine bottle in the fantasy modpack.
- Adds noodle soup.
- Reagents now have a `DATA_EXTRA_COLOR` field used for their non-base reagent overlays, for things like coloured bits in your soup. Yum! This isn't wired into any of the premade soups yet.
- Improves the contrast on pottery container sprites (and renames the overlay states to support world/inventory states, for the mug, which was missed)
- Fixes the icon for the book cart (maybe, I didn't bother to test)
- Adds Doe's fancy/damaged bookcase icons
- Adds 'decaying bookcase' loot spawners for use on Shaded Hills.
- Fixes rock, seafloor, and shrouded tiles not having footstep sounds.
- Adds an hourglass prop, with sprites by Doe.
- Rewrites SpinAnimation to avoid unnecessary list creation and to use `ANIMATION_RELATIVE`.
- Removes defunct recycler scrap code.
- Make bashing items deal damage.

## Why and what will this PR improve
A bunch of additions and fixes.

## Authorship
Ms. Adalie Desideraphobia for the code, Doe for sprites.

## Changelog
:cl: Ms. Adalie Desideraphobia
add: Added noodle soup, made by putting raw noodles in soup.
add: Added an hourglass prop.
imageadd: Added damaged sprites for bookcases.
soundadd: Readded footstep sounds for rock, seafloor, and shrouded tiles.
tweak: You can now damage items by hitting them on harm intent.
tweak: Improved the contrast on the rims of the handmade (pottery and wooden) reagent containers a little.
/:cl:

### CODE REVIEW

File: code/game/objects/items/__item.dm [hunk 1]
```diff
@@ -1219,6 +1219,6 @@ modules/mob/living/human/life.dm if you die, you will be zoomed out.
 	for(var/reagent_type in reagents.reagent_volumes)
 		var/decl/material/reagent = GET_DECL(reagent_type)
 		var/modified_reagent_overlay = state_prefix ? "[state_prefix]_[reagent.reagent_overlay]" : reagent.reagent_overlay
-		if(modified_reagent_overlay && check_state_in_icon(modified_reagent_overlay, icon))
-			reagent_overlay.overlays += overlay_image(icon, modified_reagent_overlay, reagent.get_reagent_color(), RESET_COLOR | RESET_ALPHA)
+		if(reagent.reagent_overlay && check_state_in_icon(modified_reagent_overlay, icon))
```
Comment (MistakeNot4892) [inline code/game/objects/items/__item.dm]: Should probably check this before doing the ternary/string generation above, in theory. Non-blocking.

File: code/modules/clothing/neck/jewelry.dm [hunk 2]
```diff
@@ -17,6 +18,19 @@
 /obj/item/clothing/neck/necklace/prayer_beads/basalt
 	material = /decl/material/solid/stone/basalt
 
+/obj/item/clothing/neck/necklace/prayer_beads/random
+	var/list/random_materials = list(
```
Comment (MistakeNot4892) [inline code/modules/clothing/neck/jewelry.dm:22]: `static`?

Comment (noelle-lavenza) [reply code/modules/clothing/neck/jewelry.dm:22]: No, I was considering adding a subtype with more expensive materials like gold and silver, just couldn't decide on which materials to use so I wound up not adding it. Still desirable in the future, though

File: code/game/objects/items/_item_damage.dm [hunk 3]
```diff
@@ -48,6 +48,13 @@
 	. = ..()
 	take_damage(explosion_severity_damage(severity), BURN, DAM_EXPLODE | DAM_DISPERSED, "explosion")
 
+/obj/item/bash(obj/item/weapon, mob/user)
+	. = ..()
+	var/force = weapon.get_attack_force(user)
+	if(force >= 3 && .)
```
Comment (MistakeNot4892) [inline code/game/objects/items/_item_damage.dm:54]: Minor opt, but could check `.` before retrieving force.

Comment (noelle-lavenza) [reply code/game/objects/items/_item_damage.dm:54]: My main issue with that was the double-nested indentation :(

**Comment by PlayerDeer** [comment]
This doe person sure is making a lot of sprites right now.... imagine...

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**